lib/spack/spack/cmd/release_jobs.py

```diff
@@ -452,49 +365,50 @@ def append_dep(s, d):
     return deps_json_obj
 
 
-def release_jobs(parser, args):
-    share_path = os.path.join(spack_root, 'share', 'spack', 'docker')
-    os_container_mapping_path = os.path.join(
-        share_path, 'os-container-mapping.yaml')
+def spec_matches(spec, match_string):
+    return spec.satisfies(match_string)
 
-    with open(os_container_mapping_path, 'r') as fin:
-        os_container_mapping = syaml.load(fin)
 
-    try:
-        validate(os_container_mapping, mapping_schema)
-    except ValidationError as val_err:
-        tty.error('Ill-formed os-container-mapping configuration object')
-        tty.error(os_container_mapping)
-        tty.debug(val_err)
-        return
+def find_matching_config(spec, ci_mappings):
+    for ci_mapping in ci_mappings:
+        for match_string in ci_mapping['match']:
+            if spec_matches(spec, match_string):
+                return ci_mapping['runner-attributes']
+    return None
 
-    containers = os_container_mapping['containers']
 
-    if args.specs:
-        # Just print out the spec labels and all dependency edges in
-        # a json format.
-        spec_list = [Spec(s) for s in args.specs]
-        with open(args.specs_deps_output, 'w') as out:
-            compute_spec_deps(spec_list, out)
-        return
+def release_jobs(parser, args):
+    env = ev.get_env(args, 'release-jobs', required=True)
+    env.concretize(force=args.force)
 
-    current_system = sys_type() if args.resolve_deps_locally else None
+    # FIXME: What's the difference between one that opens with 'spack'
+    # and one that opens with 'env'?  This will only handle the former.
+    yaml_root = env.yaml['spack']
 
-    release_specs_path = args.spec_set
-    if not release_specs_path:
-        raise SpackError('Must provide path to release spec-set')
+    if 'gitlab-ci' not in yaml_root:
+        tty.die('Environment yaml does not have "gitlab-ci" section')
 
-    release_spec_set = CombinatorialSpecSet.from_file(release_specs_path)
+    ci_mappings = yaml_root['gitlab-ci']['mappings']
 
-    mirror_url = args.mirror_url
+    ci_cdash = yaml_root['cdash']
```

Comment:
Will currently lead to `==> Error: 'cdash'` if the `cdash:` object is not defined in the environment.

Reply:
Glad you're trying it out @paulbry, thanks.  I'm aware of this issue and it's fixed in my current working branch, which I'll bring over to this PR sometime today.  The new work is mostly about bootstrapping compilers, which is optionaly and you should be able to ignore.